homeassistant/components/sensor/lacrosse.py

```diff
@@ -0,0 +1,231 @@
+        """Get the latest data."""
+        pass
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attributes = {}
+        attributes['low_battery'] = self._low_battery
+        attributes['new_battery'] = self._new_battery
+        return attributes
+
+    def _callback_lacrosse(self, lacrosse_sensor, user_data):
+        """Callback function that is called from pylacrosse with new values."""
+        if self._expire_after is not None and self._expire_after > 0:
+            # Reset old trigger
+            if self._expiration_trigger:
+                self._expiration_trigger()
+                self._expiration_trigger = None
+
+            # Set new trigger
+            expiration_at = (
+                dt_util.utcnow() + timedelta(seconds=self._expire_after))
+
+            self._expiration_trigger = async_track_point_in_utc_time(
+                self.hass, self.value_is_expired, expiration_at)
+
+        self._temperature = round(lacrosse_sensor.temperature * 2) / 2
+        self._humidity = lacrosse_sensor.humidity
+        self._low_battery = lacrosse_sensor.low_battery
+        self._new_battery = lacrosse_sensor.new_battery
+
+    @callback
+    def value_is_expired(self, *_):
+        """Triggered when value is expired."""
+        self._expiration_trigger = None
+        self._value = STATE_UNKNOWN
+        self.async_schedule_update_ha_state()
+
+
+class LaCrosseTemperature(LaCrosseSensor):
+    """Implementation of a Lacrosse temperature sensor."""
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._temperature
+
+
+class LaCrosseHumidity(LaCrosseSensor):
+    """Implementation of a Lacrosse humidity sensor."""
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return '%'
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._humidity
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return 'mdi:water-percent'
+
+
+class LaCrosseBattery(LaCrosseSensor):
+    """Implementation of a Lacrosse battery sensor."""
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self._low_battery is None:
+            state = 'na/a'
+        elif self._low_battery is True:
+            state = 'low'
+        else:
+            state = 'ok'
+        return state
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        if self._low_battery is None:
+            icon = 'mdi:battery-unknown'
+        elif self._low_battery is True:
+            icon = 'mdi:battery-alert'
+        else:
+            icon = 'mdi:battery'
+        return icon
+
+
+TYPE_CLASSES = {
```

Comment:
Could you please move that to the file header.

Reply:
If I move this to the beginning of the file I get the following error;
NameError: name 'LaCrosseTemperature' is not defined.
So that was the reason to keep the dict at the end. Any suggestions how to handle this?